river/ensemble/streaming_random_patches.py

```diff
@@ -0,0 +1,478 @@
+import itertools
+import collections
+import copy
+import typing
+
+import numpy as np
+
+from river import base
+from river.drift import ADWIN
+from river.metrics import Accuracy
+from river.metrics.base import MultiClassMetric
+from river.tree import HoeffdingTreeClassifier
+from river.utils.skmultiflow_utils import check_random_state
+
+
+class SRPClassifier(base.WrapperMixin, base.EnsembleMixin, base.Classifier):
+    """Streaming Random Patches ensemble classifier.
+
+    The Streaming Random Patches (SRP) [^1] is an ensemble method that
+    simulates bagging or random subspaces. The default algorithm uses both
+    bagging and random subspaces, namely Random Patches. The default base
+    estimator is a Hoeffding Tree, but other base estimators can be used
+    (differently from random forest variations).
+
+    Parameters
+    ----------
+    model
+        The base estimator.
+    n_models
+        Number of members in the ensemble.
+    subspace_size
+        Number of features per subset for each classifier where `M` is the
+        total number of features.<br/>
+        A negative value means `M - subspace_size`.<br/>
+        Only applies when using random subspaces or random patches.<br/>
+        * If `int` indicates the number of features to use. Valid range [2, M]. <br/>
+        * If `float` indicates the percentage of features to use, Valid range (0., 1.]. <br/>
+        * 'sqrt' - `sqrt(M)+1`<br/>
+        * 'rmsqrt' - Residual from `M-(sqrt(M)+1)`
+    training_method
+        The training method to use.<br/>
+        * 'subspaces' - Random subspaces.<br/>
+        * 'resampling' - Resampling.<br/>
+        * 'patches' - Random patches.
+    lam
+        Lambda value for resampling.
+    drift_detector
+        Drift detector. If `None`, disables concept drift detection and the background learner.
+    warning_detector
+        Warning detector.
```

Comment:
No, they are different. They have different sensitivity to concept drift. The warning detector is slightly more sensitive to changes, thus "warning". The drift detector is less sensitive in order to avoid false detections.

Reply:
Understood!